backend/api/auth.js

```diff
@@ -3,65 +3,97 @@ const cors = require("cors");
 const bodyParser = require("body-parser");
 const bcrypt = require("bcrypt");
 const jwt = require("jsonwebtoken");
-var db = require("../test");
+var userQuery = require('../queries/user');
 const app = express();
+
 app.use(cors());
 // support parsing of application/json type post data
 app.use(bodyParser.json());
 //support parsing of application/x-www-form-urlencoded post data
 app.use(bodyParser.urlencoded({ extended: true }));
 app.use(bodyParser.urlencoded({ extended: true }));
 
+app.get('/users', function (req, res) {
+  userQuery.selectAllUsers((result) => {
+    res.json(result)
+  });
+})
+
 app.post("/api/login", function (req, res) {
   let user = {
     username: req.body.username,
   };
-  let validUser = db.users.findByUsername(user.username);
-  if (validUser) {
-    bcrypt.compare(req.body.password, validUser.password, (err, result) => {
-      if (err) console.log(err);
-      if (result) {
-        jwt.sign({ user }, "secretkey", (err, token) => {
-          res.json({
-            token,
+  // check if the username is valid
+  userQuery.checkValidUser(user.username, (result) => {
+    let validUser = result;
+    if (validUser.length) {
+      // compare input password & password in db
+      bcrypt.compare(req.body.password, validUser[0].password, (err, result) => {
+        if (err) console.log(err);
+        if (result) {
+          jwt.sign({ user }, "secretkey", (err, token) => {
+            res.json({
+              token
+            });
           });
-        });
-      }
-    });
-  } else {
-    res.json({
-      token: false,
-    });
-  }
+        } else {
+          res.json({
+            error: true
+          })
+        }
+      });
+    } else {
+      res.json({
+        token: false,
+      });
+    }
+  });
 });
 
 app.post("/api/register", function (req, res) {
-  console.log(req.body);
   let username = req.body.username;
   let email = req.body.email;
+  let user = { username };
```

Comment:
what's the point of having a username variable and then making a obj called user?

Reply:
Aight, gonna fix this & add a 404 page. Should just have `let user = {username: req.body.username}`

Reply:
Can you merge your work after your done :eyes: